cpp/src/io/parquet/page_data.cu

```diff
@@ -466,4 +468,33 @@ void __host__ DecodeSplitPageData(cudf::detail::hostdevice_span<PageInfo> pages,
   }
 }
 
+/**
+ * @copydoc cudf::io::parquet::detail::WriteOutputBufferSizesBatched
+ */
+void __host__ WriteFinalOffsetsBatched(std::vector<size_type> const& offsets,
+                                       std::vector<size_type*> const& buff_addrs,
+                                       rmm::device_async_resource_ref mr,
+                                       rmm::cuda_stream_view stream)
+{
+  // Copy offsets to device and create an iterator
+  auto d_src_data = cudf::detail::make_device_uvector_async(offsets, stream, mr);
+  auto src_iter   = cudf::detail::make_counting_transform_iterator(
```

Comment:
wait, isn't this the same as `d_src_data.begin()`?

Reply:
Since `size_iter` is an constant_iterator, we have to use counting_transform_iterators for src and dst too to avoid compilation (template) errors for `cub::DeviceMemcpy::Batched`

Reply:
Not sure what you mean. I think they can be different types https://nvidia.github.io/cccl/cub/api/structcub_1_1DeviceMemcpy.html

Reply:
Yeah, you are right. My bad about misreading the compiler output. We are ok with the `d_dst_adds.begin()` but we do need to create the `src_iter` as `d_src_data` contains actual values (scalars) and not addresses.

Reply:
All done!